app/code/Magento/SalesRule/Model/ResourceModel/Rule/Collection.php

```diff
@@ -332,13 +332,9 @@ public function addWebsiteGroupDateFilter($websiteId, $customerGroupId, $now = n
                 'customer_group_ids.' .
                 $entityInfo['entity_id_field'] .
                 ' = cgw.' .
-                $entityInfo['entity_id_field'] . ' AND ' . $websiteId . ' = cgw.website_id',
+                $entityInfo['entity_id_field'] . ' AND ' . (int) $websiteId . ' = cgw.website_id',
                 []
-            )->where(
-                'cgw.website_id IS NULL',
-                $websiteId,
-                \Zend_Db::INT_TYPE
-            );
+            )->where('cgw.website_id IS NULL');
```

Comment:
👍 Totally agree with it

Reply:
Requested changes here? Why? Just small fix for better understanding what happening here, because for first look it seems, that there are filtering via `$websiteId`

Reply:
No changes here. I'm agree with this update